libfoundation/src/foundation-foreign.cpp

```diff
@@ -280,6 +297,18 @@ static bool __uint_import(void *contents, bool release, MCValueRef& r_value)
     return MCNumberCreateWithUnsignedInteger(*(uinteger_t *)contents, (MCNumberRef&)r_value);
 }
 
+static bool
+__size_import (void *contents,
+               bool release,
+               MCValueRef & r_value)
+{
+	size_t t_value = *(size_t *) contents;
+	if (t_value > UINTEGER_MAX)
+		return false;
```

Comment:
I think you need to throw an error here - otherwise you won't see why the mapping failed at the script-level.